provider/openstack/local_test.go

```diff
@@ -580,6 +581,27 @@ func (s *localServerSuite) TestStartInstanceNetworkNotSetReturnsError(c *gc.C) {
 	or supply it via --config when creating a new model`)
 }
 
+func (s *localServerSuite) TestStartInstanceNoNetworksNetworkNotSetNoError(c *gc.C) {
+	// Modify the server that is created by default, to clear the networks.
+	model := neutronmodel.New()
+	for _, net := range model.AllNetworks() {
+		model.RemoveNetwork(net.Id)
+	}
+	s.srv.Openstack.Neutron.AddNeutronModel(model)
```

Comment:
Nice find!  I didn't think of using the NeutronModel functionality that way, might be useful other places too.